### DISCUSSION TITLE
'dvc add' behavior on Windows

### DISCUSSION THREAD

**Discussion by dfossl** [treeverse/dvc#7795] [Help]
Hello!

I have been playing around with dvc as a potential tracking tool and have some questions about the 'dvc add' behavior. I have watched the tutorial [here](https://dvc.org/doc/start/data-management) and when 
```shell
dvc add data/data.xml
```
is called it creates a file data.data.xml.dvc IN the data/ directory.
BUT
When I run the same command I get a data.xml.dvc file in my current directory that as a path variable set to data/data.xml .

Is this the expected functionality? It is quite frustrating to have the .dvc files fill up the main directory.

Is there some configuration I am missing? Is this a Windows quirk?

Also, the way 'dvc add' is functioning is that is adds the dvc file to the directory where 'dvc add' was called. This means if not careful one can have multiple dvc files for the same data. Shouldn't the dvc file always be placed where the data is, that seems saver?

Thanks for any help, not sure if I am doing something wrong or not!

**Comment by karajan1001** [comment]
Sorry I could not reproduce it on my windows computer.
Could you please show me the result of the script:
```bash
mkdir data
cd data
git init
dvc init
mkdir data
echo data > data/data.xml
dvc add data/data.xml
```

**Reply by dfossl** [reply]
> Sorry I could not reproduce it on my windows computer. Could you please show me the result of the script:
> 
> ```shell
> mkdir data
> cd data
> git init
> dvc init
> mkdir data
> echo data > data/data.xml
> dvc add data/data.xml
> ```

Ah ha! I have identified the quirk. When run the above command it works fine in my local C: drive. BUT when I run it in my "T:" drive which is a network drive I have for work it results in:
```
data/
    |-.dvc
    |-.git
    |-.dvcignore
    |-data.xml.csv
    |-data/
        |-data.xml
```
Any idea what may be the cause of this quirk in behavior between local vs, network drives?

**Comment by karajan1001** [comment]
I'm sorry I can't test network drives on my side, I can only guess it might be a bug of DVC on NFS. 
Could you please give some more info? The result of `dvc doctor` on `T:`, thank you.

**Comment by dfossl** [comment]
Here ya go:
```
DVC version: 2.10.2 (pip)
---------------------------------
Platform: Python 3.10.4 on Windows-10-10.0.19042-SP0
Supports:
        webhdfs (fsspec = 2022.3.0),
        http (aiohttp = 3.8.1, aiohttp-retry = 2.4.6),
        https (aiohttp = 3.8.1, aiohttp-retry = 2.4.6)
Cache types:
Cache directory: NTFS on T:\
Caches: local
Remotes: None
Workspace directory: NTFS on T:\
Repo: dvc, git
```
Thanks for the help! If I have time I will try and follow the 'add' command execution path through to see where it may be going wrong.
